kombu/connection.py

```diff
@@ -844,6 +844,8 @@ def _extract_failover_opts(self):
                 conn_opts['interval_step'] = transport_opts['interval_step']
             if 'interval_max' in transport_opts:
                 conn_opts['interval_max'] = transport_opts['interval_max']
+            if 'timeout' in transport_opts:
+                conn_opts['timeout'] = transport_opts['timeout']                
```

Comment:
I prefer to use different name for 'timeout' in `transport_opts` This `timeout` value is overall timeout for all retries:
https://github.com/celery/kombu/blob/56e88dc275831196ca396fce81275ac687d1207b/kombu/utils/functional.py#L322
. We have also `connect_timeout` which specifies timeout of creating connection to single broker.
https://github.com/celery/kombu/blob/56e88dc275831196ca396fce81275ac687d1207b/kombu/connection.py#L81

If not good name is picked, it will be confusing for users.

Reply:
I agree that it is confusing, because I was confused investigating :)

The current references to a timeout in connection.py:

1. https://github.com/celery/kombu/blob/56e88dc275831196ca396fce81275ac687d1207b/kombu/connection.py#L313: Why is there a reference in the documentation? Because it may be in kwargs? Would it be more suitable to reference transport docs for forwarded possible parameters?

2. https://github.com/celery/kombu/blob/56e88dc275831196ca396fce81275ac687d1207b/kombu/connection.py#L390: This references the same timeout as it is forwarded. 

So my suggestion would be to call it `connection_timeout` in the transport options. Thoughts?

Reply:
I renamed it accordingly, add also added more functionality where I thought it is usefull to use the new option. Please check